File: crates/sdk/src/artifact/staticcheck.rs
use crate::{
    artifact::{language::go::GoBuilder, ArtifactSourceBuilder},
    context::ConfigContext,
};
use anyhow::Result;

pub async fn build(context: &mut ConfigContext) -> Result<String> {
    let name = "staticcheck";

    let source_digest = "e8f40ddbc450bf26d1855916519283f7c997ffedbcb971e2a7b92892650d92b6";
    let source_version = "2025.1.1";
    let source_path =
        format!("https://github.com/dominikh/go-tools/archive/refs/tags/{source_version}.tar.gz");

    let source = ArtifactSourceBuilder::new(name, source_path.as_str())
        .with_digest(source_digest)
        .build();

    let build_dir = format!("go-tools-{}", source_version);
    let build_path = format!("cmd/{}/{}.go", name, name);

    GoBuilder::new(name)
        .with_build_dir(build_dir.as_str())
        .with_build_path(build_path.as_str())
        .with_source(source)
        .build(context)
        .await
}

File: sdk/python/src/vorpal_sdk/artifact/staticcheck.py
"""staticcheck artifact (built from the go-tools source via the Go builder).

Mirrors ``sdk/typescript/src/artifact/staticcheck.js``.
"""

from __future__ import annotations

from typing import TYPE_CHECKING

from vorpal_sdk.artifact import ArtifactSource
from vorpal_sdk.system import SYSTEMS

if TYPE_CHECKING:
    from vorpal_sdk.context import ConfigContext


class Staticcheck:
    """Builder for the staticcheck artifact."""

    def build(self, context: ConfigContext) -> str:
        # Deferred import breaks the go-tool<->language-builder cycle.
        from vorpal_sdk.artifact.language.go import Go

        name = "staticcheck"
        source_version = "2026.1"
        source_path = (
            f"https://sdk.vorpal.build/source/"
            f"staticcheck-{source_version}.tar.gz"
        )

        source = ArtifactSource(name, source_path).build()

        build_directory = f"go-tools-{source_version}"
        build_path = f"cmd/{name}/{name}.go"

        return (
            Go(name, SYSTEMS)
            .with_aliases([f"{name}:{source_version}"])
            .with_build_directory(build_directory)
            .with_build_path(build_path)
            .with_source(source)
            .build(context)
        )
